### PR TITLE
fix(floatwin): don't always use winheight

### LINKED ISSUE #13298: When `'winheight'` is set, `nvim_open_win()` does not work well.
- `nvim --version`: `NVIM v0.5.0-837-g0798ad3a3`
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: `Ubuntu 18.04` (WSL2)
- Terminal name/version: [wsltty](https://github.com/mintty/wsltty)
- `$TERM`: `xterm-256color`

### Steps to reproduce using `nvim -u NORC`

```vim
" minimal_vimrc
set winheight=20

let opts = {
\  'height': 1,
\  'col': 5,
\  'row': 1,
\  'relative': 'editor',
\  'style': 'minimal',
\  'width': 15
\}

let bufnr = nvim_create_buf(v:false, v:true)
call nvim_buf_set_lines(bufnr, 0, -1, v:true, ['hello world'])
let win_id = nvim_open_win(bufnr, v:true, opts)
```

1. `nvim -u NORC`
2. `:so minimal_vimrc`

### Actual behaviour

The height of the floating window changes to the 20 specified in `set winheight=20`, instead of the 1 in `height` passed in the `nvim_open_win()` argument.

### Expected behaviour

The height of the floating window opened by `nvim_open_win()` should be the height of `height` specified by the argument.

---

I encountered this when I was using https://github.com/nvim-telescope/telescope.nvim.
Is this behavior correct?

### PR DESCRIPTION
**Pull request by vigoux**
Fixes #13298 and add a test for it, hopefully this patch is not too dumb.

### CODE REVIEW

File: test/functional/ui/float_spec.lua [hunk 1]
```diff
@@ -83,6 +83,28 @@ describe('floatwin', function()
     assert_alive()
   end)
 
+  it('opened with correct height', function()
```
Comment (bfredl) [inline test/functional/ui/float_spec.lua:86]: perhaps also write the same test for winwidth while we are at it

Comment (vigoux) [reply test/functional/ui/float_spec.lua:86]: Sounds good

Comment (vigoux) [reply test/functional/ui/float_spec.lua:86]: done

**Comment by vigoux** [comment]
@bfredl is it okay for you now ?

**Pull request merged**